packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -880,6 +876,82 @@ export class BaseResolversVisitor<
+        const referencedTypes = schemaType.getTypes().map(unionMemberType => {
+          const isUnionMemberMapped = this.config.mappers[unionMemberType.name];
+
+          // 1. If mapped without plachoder, just use it without doing extra checks
+          if (isUnionMemberMapped && !hasPlaceholder(isUnionMemberMapped.type)) {
+            return isUnionMemberMapped.type;
+          }
+
+          // 2. Work out value for union member type
+          // 2a. By default, use the typescript type
+          let unionMemberValue = this.convertName(unionMemberType.name);
+
+          // 2b. Find fields to Omit if needed.
+          //  - If no field to Omit, "type with maybe Omit" is typescript type i.e. no Omit
+          //  - If there are fields to Omit, "type with maybe Omit"
+          const fields = unionMemberType.getFields();
+          const fieldsToOmit = this._federation
+            .filterFieldNames(Object.keys(fields))
+            .map(fieldName => {
+              const field = fields[fieldName];
+              const baseType = getBaseType(field.type);
+              const isMapped = this.config.mappers[baseType.name];
+              const isUnion = isUnionType(baseType);
+              const shouldMapType = this._shouldMapType[baseType.name];
+
+              if (!isMapped && !isUnion && !shouldMapType) {
+                return null;
+              }
+
+              return {
+                addOptionalSign: !this.config.avoidOptionals && !isNonNullType(field.type),
+                fieldName: field.name,
+                replaceWithType: wrapTypeWithModifiers(this.getTypeToUse(baseType.name), field.type, {
+                  wrapOptional: this.applyMaybe,
+                  wrapArray: this.wrapWithArray,
+                }),
+              };
+            })
+            .filter(a => a);
+          if (fieldsToOmit.length > 0) {
+            unionMemberValue = this.replaceFieldsInType(unionMemberValue, fieldsToOmit);
+          }
+
+          // 2c. If union member is mapped with placeholder, use the "type with maybe Omit" as {T}
+          if (isUnionMemberMapped && hasPlaceholder(isUnionMemberMapped.type)) {
+            return replacePlaceholder(isUnionMemberMapped.type, unionMemberValue);
+          }
+
+          // 2d. If has default mapper with placeholder, use the "type with maybe Omit" as {T}
+          const hasDefaultMapper = !!this.config.defaultMapper?.type;
+          const isScalar = this.config.scalars[typeName];
+
+          if (hasDefaultMapper && hasPlaceholder(this.config.defaultMapper.type)) {
+            const finalTypename = isScalar ? this._getScalar(typeName) : unionMemberValue;
+            return replacePlaceholder(this.config.defaultMapper.type, finalTypename);
+          }
+
+          return unionMemberValue;
+        });
+        res[typeName] = referencedTypes.map(type => `( ${type} )`).join(' | '); // Must wrap every union member in explicit "( )" to separate the members
```

Comment:
> Must wrap every union member in explicit "( )" to separate the members

Result usually looks like this:
```ts
export type ResolversUnionTypes =  {
  UnionThing: ( TypeA ) | ( TypeB )
}
```
Prettier should remove the round brackets automatically i.e. result will be `UnionThing: TypeA | TypeB`

I've added it this way so it's a bit easier to add types to the union members without ambiguity e.g.
```ts
UnionThing: TypeA & { __typename: 'TypeA' } | TypeB & { __typename: 'TypeB' } // This is ambiguous

// Prettier will try to turn it into this anyways:
UnionThing: ( TypeA & { __typename: 'TypeA' } ) | ( TypeB & { __typename: 'TypeB' } ) 
```